storage_service/locations/models/dspace_rest.py

```diff
@@ -0,0 +1,379 @@
+"""
+Integration with DSpace, using REST API as the protocol.
+
+NOTE THAT IN ALL REQUESTS TO THE REST API THE SSL VERIFICATION HAS BEEN DISABLED
+
+"""
+from __future__ import absolute_import
+# stdlib, alphabetical
+import json
+import logging
+import os
+import shutil
+import subprocess
+import urllib
+
+# Core Django, alphabetical
+from django.db import models
+from django.utils.translation import ugettext as _, ugettext_lazy as _l
+
+# Third party dependencies, alphabetical
+from lxml import etree
+import requests
+
+# This project, alphabetical
+from agentarchives import archivesspace
+
+# This module, alphabetical
+from common import utils
+from .location import Location
+
+LOGGER = logging.getLogger(__name__)
+
+
+class DSpaceREST(models.Model):
+    """Integration with DSpace using the REST API."""
+    space = models.OneToOneField('Space', to_field='uuid')
+    rest_url = models.URLField(max_length=256,  # ATTENTION could be with or without slash at end
+                               verbose_name=_l("REST URL "),
+                               help_text=_l('URL of the REST API. E.g. http://demo.dspace.org/rest'))
+    user = models.CharField(max_length=64,
+                            verbose_name=_l("User"),
+                            help_text=_l('DSpace username to authenticate as'))
+    password = models.CharField(max_length=64,
+                                verbose_name=_l("Password"),
+                                help_text=_l('DSpace password to authenticate with'))
+
+    ARCHIVE_FORMAT_ZIP = 'ZIP'
+    ARCHIVE_FORMAT_7Z = '7Z'
+    ARCHIVE_FORMAT_CHOICES = (
+        (ARCHIVE_FORMAT_ZIP, 'ZIP'),
+        (ARCHIVE_FORMAT_7Z, '7z'),
+    )
+
+    archive_format = models.CharField(max_length=3,
+                                      choices=ARCHIVE_FORMAT_CHOICES,
+                                      default=ARCHIVE_FORMAT_ZIP,
+                                      verbose_name=_l('Archive format'))
```

Comment:
Also, is `archive_format` used at all? If not, is it necessary?

Reply:
Also, is `archive_format` used at all? If not, is it necessary?

Reply:
This is a "legacy" issue from the DSpace location. I hadn't removed it because I hadn't thought clearly about it.

What the DSpace location does is it takes the zipped AIP and splits it into two zips. So yeah, it's unnecessary.